File: windows/Scanner/Rules/AESAUTH001.cs
using AIExposureScanner.Scanner.Models;

namespace AIExposureScanner.Scanner.Rules;

public sealed class AESAUTH001 : IRule
{
    public string Id => "AES-AUTH-001";
    public Severity Severity => Severity.Critical;

    public IReadOnlyList<Finding> Evaluate(ScanFacts facts)
    {
        var findings = new List<Finding>();

        foreach (var server in facts.McpServers.Where(server => !server.Disabled))
        {
            findings.AddRange(
                server.Env.Values
                    .Where(SecretPatterns.ContainsSecret)
                    .Select(value => this.Finding(
                        server.AppId,
                        server.Name,
                        affectedPath: server.ConfigPath,
                        maskedValue: SecretPatterns.Masked(value)
                    ))
            );
        }

        findings.AddRange(
            facts.Settings
                .Where(setting => setting.Key == "apiKey")
                .Where(setting => setting.StringValue is not null && SecretPatterns.ContainsSecret(setting.StringValue))
                .Select(setting => this.Finding(
                    setting.AppId,
                    affectedPath: setting.ConfigPath,
                    maskedValue: SecretPatterns.Masked(setting.StringValue!)
                ))
        );

        return findings;
    }
}

File: macos/Scanner/Sources/Scanner/Rules/AESAUTH001.swift
public struct AESAUTH001: Rule {
    public let id = "AES-AUTH-001"
    public let severity: Severity = .critical

    public init() {}

    public func evaluate(_ facts: ScanFacts) -> [Finding] {
        var findings: [Finding] = []

        for server in facts.mcpServers where !server.disabled {
            for value in server.env.values where SecretPatterns.containsSecret(value) {
                findings.append(
                    finding(
                        app: server.appId,
                        serverName: server.name,
                        affectedPath: server.configPath,
                        maskedValue: SecretPatterns.masked(value)
                    )
                )
            }
        }

        for setting in facts.settings where setting.key == "apiKey" {
            if let value = setting.stringValue, SecretPatterns.containsSecret(value) {
                findings.append(
                    finding(
                        app: setting.appId,
                        affectedPath: setting.configPath,
                        maskedValue: SecretPatterns.masked(value)
                    )
                )
            }
        }

        return findings
    }
}
